internals/overlord/cmdstate/request.go

```diff
@@ -70,8 +70,16 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		return nil, ExecMetadata{}, errors.New("cannot use interactive mode without a terminal")
 	}
 
-	// Inherit the pebble daemon environment.
+	// Inherit the pebble daemon environment
+	// If the user is being changed, reset the HOME and USER env vars so that they
+	// can be set correctly later on in this method
```

Comment:
Punctuation nits:

```suggestion
	// Inherit the pebble daemon environment.
	// If the user is being changed, unset the HOME and USER env vars so that they
	// can be set correctly later on in this method.
```